src/App.tsx

```diff
@@ -1,14 +1,77 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { FilterOptions } from './types/FilterOptions';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+
+function getFilteredTodos(
+  todos: Todo[],
+  filterOption: FilterOptions,
+  query: string,
+): Todo[] {
+  const normalizedQuery = query.toLowerCase().trim();
+
+  let filteredTodos: Todo[] = todos.filter(todo => {
+    switch (filterOption) {
+      case FilterOptions.Active:
+        return !todo.completed;
+
+      case FilterOptions.Completed:
+        return todo.completed;
+
+      default:
+        return true;
+    }
+  });
+
+  filteredTodos = filteredTodos.filter(todo => {
+    const normalisedTodoTitle = todo.title.toLowerCase().trim();
+
+    return normalisedTodoTitle.includes(normalizedQuery);
+  });
+
+  return filteredTodos;
+}
```

Comment:
This function `getFilteredTodos` is defined outside of the App component. It's better to move this function inside the component and use `useCallback` hook to memoize it. This way, the function will not be recreated on each render which will improve the performance of your app.